src/transformers/models/bert/modeling_bert.py

```diff
@@ -192,10 +203,22 @@ def forward(
         seq_length = input_shape[1]
 
         if position_ids is None:
-            position_ids = self.position_ids[:, past_key_values_length : seq_length + past_key_values_length]
+            if hasattr(self, "position_ids"):
+                position_ids = self.position_ids[:, past_key_values_length : seq_length + past_key_values_length]
+            else:
+                position_ids = torch.arange(seq_length).expand((1, -1))
 
         if token_type_ids is None:
-            token_type_ids = torch.zeros(input_shape, dtype=torch.long, device=self.position_ids.device)
+            if hasattr(self, "token_type_ids"):
+                token_type_ids = self.token_type_ids[:, :seq_length]
+
+        # Setting the token_type_ids to the registered buffer in constructor where it is all zeros, which usually occurs
+        # when its auto-generated, registered buffer helps users when tracing the model without passing token_type_ids, solves
+        # issue #5664
+
+        elif token_type_ids is not None and len(torch.nonzero(token_type_ids)) < 1:
+            if hasattr(self, "token_type_ids"):
+                token_type_ids = self.token_type_ids[:, :seq_length]
```

Comment:
Why is this necessary?

Reply:
@LysandreJik it was a workaround to find if token_type_ids are autogenerated as they were not None at this point (auto-generated token_type_ids are all zeros), this way when its auto-generated we could use the buffer instead and if user pass it (non all zero) then we don't need to handle it as it works fine. 
From the comments above seems checking if hasattr(self.embeddings, "token_type_ids") might be a better way to find out if its was auto-generated?